test-utils/src/base-fake-repository.ts

```diff
@@ -1,80 +1,80 @@
-const { validate } = require('class-validator');
+import { validate, ValidationError } from 'class-validator';
 
-export class BaseFakeRepository {
-  #__rows: Array<any> = [];
-  #__ModelClass: any;
-
-  constructor(ModelClass: any) {
-    this.#__ModelClass = ModelClass;
-  }
+interface BaseModel {
+  [key: string]: any;
+}
 
-  async create(args: any) {
-    const model = await this.__initModel(args.data);
-    this.#__rows.push(model);
+export class BaseFakeRepository<T extends BaseModel> {
+  private rows: T[] = [];
+  private ModelClass: new () => T;
 
-    return __deepCopy(model, args.select);
+  constructor(ModelClass: new () => T) {
+    this.ModelClass = ModelClass;
   }
 
-  async createUnscoped(args: any) {
-    const model = await this.__initModel(args.data);
-    this.#__rows.push(model);
-
-    return __deepCopy(model, args.select);
+  async create(args: { data: Partial<T>; select?: Record<string, boolean> }): Promise<T> {
+    const model = await this.initModel(args.data);
+    this.rows.push(model);
+    return this.deepCopy(model, args.select);
   }
 
-  async __initModel(data: any) {
-    const model = new this.#__ModelClass();
-    Object.assign(model, data);
-
-    const errors = await validate(model);
-    if (errors.length) throw new Error('\n' + errors);
-
-    return model;
+  async findMany(args: { where: Partial<T>; select?: Record<string, boolean> }): Promise<T[]> {
+    const rows = this.rows.filter(row => this.matchWhereCondition(row, args.where));
+    return rows.map(row => this.deepCopy(row, args.select));
   }
 
-  async findMany(args: any) {
-    return this.__findMany(args).map(row => __deepCopy(row, args.select));
+  async findById(id: string): Promise<T> {
+    const row = this.findByIdInternal(id);
+    return this.deepCopy(row);
   }
 
-  __findMany(args: any) {
-    return this.#__rows.filter(row => {
-      for (const key in args.where) {
-        if (row[key] !== args.where[key]) return false;
-      }
-      return true;
-    });
+  async updateById(id: string, args: { data: Partial<T> }): Promise<T> {
+    const row = this.findByIdInternal(id);
+    Object.assign(row, args.data);
+    return this.deepCopy(row);
   }
 
-  __findById(id: string) {
-    const rows = this.__findMany({ where: { id: id } });
-    if (rows.length === 0) throw new Error(`could not find id ${id}`);
-    return rows[0];
+  private async initModel(data: Partial<T>): Promise<T> {
```

Comment:
Regardless of the `private` keyword our convention for private is reflected in names by using `__`. Please revert removal of underscores.